ts/screens/wallet/payment/ManualDataInsertionScreen.tsx

```diff
@@ -149,54 +147,53 @@ class ManualDataInsertionScreen extends React.Component<Props, State> {
             </TouchableDefaultOpacity>
 
             <Form>
-              <Item
-                style={styles.noLeftMargin}
-                floatingLabel={true}
-                error={this.state.paymentNoticeNumber
-                  .map(isLeft)
-                  .getOrElse(false)}
-                success={this.state.paymentNoticeNumber
-                  .map(isRight)
-                  .getOrElse(false)}
-              >
-                <Label>{I18n.t("wallet.insertManually.noticeCode")}</Label>
-                <Input
-                  keyboardType={"numeric"}
-                  returnKeyType={"done"}
-                  maxLength={18}
-                  onChangeText={value => {
+              <LabelledItem
+                type="text"
+                isValid={this.state.paymentNoticeNumber
+                  .map<true | false | undefined>(e =>
+                    e.fold(
+                      _ => false,
+                      _ => true
+                    )
+                  )
+                  .getOrElse(undefined)}
+                label={I18n.t("wallet.insertManually.noticeCode")}
+                inputProps={{
+                  keyboardType: "numeric",
+                  returnKeyType: "done",
+                  maxLength: 18,
+                  onChangeText: value => {
                     this.setState({
                       paymentNoticeNumber: some(value)
                         .filter(NonEmptyString.is)
                         .map(_ => PaymentNoticeNumberFromString.decode(_))
                     });
-                  }}
-                />
-              </Item>
-              <Item
-                style={styles.noLeftMargin}
-                floatingLabel={true}
-                error={this.state.organizationFiscalCode
-                  .map(isLeft)
-                  .getOrElse(false)}
-                success={this.state.organizationFiscalCode
-                  .map(isRight)
-                  .getOrElse(false)}
-              >
-                <Label>{I18n.t("wallet.insertManually.entityCode")}</Label>
-                <Input
-                  keyboardType={"numeric"}
-                  returnKeyType={"done"}
-                  maxLength={11}
-                  onChangeText={value => {
+                  }
+                }}
+              />
+              <LabelledItem
+                type="text"
+                isValid={
+                  isNone(this.state.organizationFiscalCode)
+                    ? undefined
+                    : this.state.organizationFiscalCode
+                        .map(isRight)
+                        .getOrElse(false)
```

Comment:
This is very similar to [isValid above](https://github.com/pagopa/io-app/pull/3143/files#diff-2b05a8853c86e3951e05baeb33fc4702ac7809c1e1f87e4ee44e82c1e21160e6R152): since you went for a change there, I suggest you refactor this part as well to keep the style very consistent and easy to review.

Seeing how popular this mapping seems to be in our codebase, it might make sense to extract it to a separate helper in the (far) future. Something along the lines of `validateOptionalEither<T>(optionOfEither: Option<Either<T>>): true | false | undefined`. Food for thought 🍰